engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -75,66 +88,13 @@ class PartwiseJoin : public kernel {
-	std::unique_ptr<ral::frame::BlazingTable> load_set(BatchSequence & input, std::shared_ptr<ral::cache::CacheMachine> cache, bool load_all){
-		std::size_t bytes_loaded = 0;
-		std::vector<std::unique_ptr<ral::frame::BlazingTable>> tables_loaded;
-		if (input.wait_for_next()){
-			tables_loaded.emplace_back(input.next());
-			bytes_loaded += tables_loaded.back()->sizeInBytes();
-		} else {
-			return nullptr;
-		}
-		// NOTE this used to be like this:
-		// while ((input.has_next_now() && bytes_loaded < join_partition_size_theshold) ||
-		//  	(load_all && input.wait_for_next())) {
-		// with the idea that it would just start processing as soon as there was data to process.
-		// This actually does not make it faster, because it makes it so that there are more chunks to do pairwise joins and therefore more join operations
-		// We may want to revisit or rethink this
-
-		while ((input.wait_for_next() && bytes_loaded < join_partition_size_theshold) ||
-					(load_all && input.wait_for_next())) {
-			tables_loaded.emplace_back(std::move(input.next()));
-			bytes_loaded += tables_loaded.back()->sizeInBytes();
-
-			//If the concatenation so far produces a string overflow, we put back the batch because in this case the order of the elements does not matter
-			if(!load_all && ral::utilities::checkIfConcatenatingStringsWillOverflow(tables_loaded)){
-				bytes_loaded -= tables_loaded.back()->sizeInBytes();
-				cache->addToCache(std::move(tables_loaded.back()));
-				tables_loaded.pop_back();				
-				break;
-			}
-		}
-
-		if (tables_loaded.size() == 0){
-			return nullptr;
-		} else if (tables_loaded.size() == 1){
-			return std::move(tables_loaded[0]);
-		} else {
-			std::vector<ral::frame::BlazingTableView> tables_to_concat(tables_loaded.size());
-			for (std::size_t i = 0; i < tables_loaded.size(); i++){
-				tables_to_concat[i] = tables_loaded[i]->toBlazingTableView();
-			}
-
-			// if we were supposed to load all and strings will overflow, lets at least provide a warning
-			if (load_all && ral::utilities::checkIfConcatenatingStringsWillOverflow(tables_to_concat)){
-				logger->warn("{query_id}|{step}|{substep}|{info}",
-								"query_id"_a=(context ? std::to_string(context->getContextToken()) : ""),
-								"step"_a=(context ? std::to_string(context->getQueryStep()) : ""),
-								"substep"_a=(context ? std::to_string(context->getQuerySubstep()) : ""),
-								"info"_a="In PartwiseJoin::load_set Concatenating Strings will overflow strings length");			
-			}
-
-			return ral::utilities::concatTables(tables_to_concat);
-		}
-	}
-
-
     std::unique_ptr<ral::frame::BlazingTable> load_left_set(){
 
 		this->max_left_ind++;
-		// need to load all the left side, only if doing a FULL OUTER JOIN
-		bool load_all = this->join_type == OUTER_JOIN ?  true : false;
-		auto table = load_set(this->left_sequence, this->input_.get_cache("input_a"), load_all);
+		std::unique_ptr<ral::frame::BlazingTable> table;
+		if (this->left_sequence.wait_for_next()){
```

Comment:
why isnt this one operation like get_or_wait?

Reply:
good point. Let fix that.

Reply:
well, btw get_or_wait is for array like access. I am just going to do `next()` since it will wait or return null if there are none